shared/js/baseOptions.js

```diff
@@ -0,0 +1,149 @@
+startTiming('Puzzle', script_path, '');
```

Comment:
should this file maybe have a comment at the top explaining what it is? And maybe have a name that indicates that it has something to do with "apps".

It also kind of does things (and not just related to options), such as "startTiming"...